test/images/sample-device-plugin/sampledeviceplugin.go

```diff
@@ -200,3 +230,20 @@ func handleRegistrationProcess(registerControlFile string, dpStub *plugin.Stub,
 		}
 	}
 }
+
+func createCDIFile(devs []*pluginapi.Device) error {
+	content := fmt.Sprintf(`{"cdiVersion":"%s","kind":"%s","devices":[`, cdiVersion, resourceName)
+	for i, dev := range devs {
+		name := cdiPrefix + dev.ID
+		content += fmt.Sprintf(`{"name":"%s","containerEdits":{"env":["CDI_DEVICE=%s"],"deviceNodes":[{"path":"/tmp/%s","type":"b","major":1,"minor":%d}]}}`, name, name, name, i)
```

Comment:
Not critical, but it may be easier to read if we construct the `envvar` and `deviceNode` "snippets separately.

Would using `strings.Builder` here instead be an option?

Reply:
I don't think strings.Builder will help here. Marshalling into JSON would be better, but more verbose. [text.template](https://pkg.go.dev/text/template) wouldn't help much either in my opinion.

Reply:
Here is an attempt to use marshalling: https://go.dev/play/p/NPcK09ovbI5
Too verbose to my taste, considering a purpose. I'd propose to leave it as it is.

Reply:
@elezar If [cdi spec module](https://github.com/cncf-tags/container-device-interface/blob/main/specs-go/config.go) could be imported without bringing tons of unwanted deps, I'd just import it and that would be the best solution.

Reply:
This isn't a blocker at all. As you say ... having fewer dependencies from the CDI spec would be better.